src/helpers.rs

```diff
@@ -512,6 +516,39 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         let value_place = op_place.offset(offset, MemPlaceMeta::None, layout, this)?;
         this.write_scalar(value, value_place.into())
     }
+
+    /// Parse a `timespec` struct and return it as a `std::time::Duration`. The outer `Result` is
+    /// for interpreter errors encountered while reading memory, and the inner `Result` indicates
+    /// whether the value in the `timespec` struct is invalid. Some libc functions will return
+    /// `EINVAL` if the struct's value is invalid.
+    fn read_timespec(
+        &mut self,
+        timespec_ptr_op: OpTy<'tcx, Tag>,
+    ) -> InterpResult<'tcx, Result<Duration, TimespecError>> {
```

Comment:
We don't usually use the "ZST error type pattern" in Miri, and I think just returning an `Option<Duration>` would be clearer here.

Reply:
Then you can even use `.try_into()?` :D